rust/operator/src/hdfs_controller.rs

```diff
@@ -60,11 +61,18 @@ pub async fn reconcile_hdfs(hdfs: Arc<HdfsCluster>, ctx: Arc<Ctx>) -> HdfsOperat
     let namenode_podrefs = hdfs.pod_refs(&HdfsRole::NameNode)?;
     let journalnode_podrefs = hdfs.pod_refs(&HdfsRole::JournalNode)?;
 
+    let mut cluster_resources = ClusterResources::new(
+        APP_NAME,
+        RESOURCE_MANAGER_HDFS_CONTROLLER,
+        &hdfs.object_ref(&()),
+    )
+    .map_err(|e| Error::CreateClusterResources { source: e })?;
+
     let discovery_cm = build_discovery_configmap(&hdfs, &namenode_podrefs)
         .map_err(|e| Error::BuildDiscoveryConfigMap { source: e })?;
 
-    client
-        .apply_patch(FIELD_MANAGER_SCOPE, &discovery_cm, &discovery_cm)
+    cluster_resources
+        .add(client, &discovery_cm)
```

Comment:
As discussed, this is dependant on the cluster definition and thus can never become an orphan (or even stale).

Reply:
done